### ISSUE TITLE
Remove "independent" rubygems package from build

### ISSUE THREAD

**Issue by voxik** [minimization/content-resolver-input#165]
Looking at [[1]], rubygems package is marked as unwanted but available. The problem with this packages is, that in Fedora, there is rubygems package [[2]], but there is also rubygems subpackage of ruby package [[3]], which should be enough to satisfy all ELN needs. Please drop the rubygems package and  improve the content-resolver to properly handle scenarios like this.

[1]: https://tiny.distro.builders/view-unwanted--view-eln--x86_64.html
[2]: https://src.fedoraproject.org/rpms/rubygems
[3]: https://src.fedoraproject.org/rpms/ruby/blob/master/f/ruby.spec#_262

**Comment by voxik** [comment]
Ping! Any update seeing this package still listed here:

https://tiny.distro.builders/view-rpm--view-eln--rubygems.html

**Comment by tdawson** [comment]
Ohhh ... I just got that.  I misunderstood thinking you wanted the binary rubygems removed (which I didn't know how) but you are meaning to just not have the source.rpm.  It's a little tricky, but we have something similar with kernel-headers.

**Comment by voxik** [comment]
Yes, right, I want to remove the SRPM, because binary RPM is provided anyway.

**Comment by tdawson** [comment]
OK.  I have done the same that I did for kernel-headers, and it's seeming to work pretty good.
I have put it in the ELN building exclude list, with a comment on why.  This will keep ELN from building it, even if it shows up as a srpm in Content Resolver.
I have untagged all builds of rubygems from eln, in koji.  This will keep the rubygems srpm from being in the ELN repo.  This *should* eventually cause Content-Resolver and buildroot generator to quit pulling it in.
It takes a little while for the koji -> ELN repo -> Content Resolver path to flow.  So I'll check on it tomorrow, and make sure we only have the binaries left in Content Resolver.

**Comment by voxik** [comment]
Thx. Could you please share some references to commits or some commands etc or is it some black magic one should not know? :innocent:

**Comment by tdawson** [comment]
You'd have to have permissions.  So not everyone could do it.
- Excludes (this will be changing at some point to a general config file someplace)
https://github.com/fedora-ci/eln-periodic/commit/9d0a5afd7c383c0232379c1c69f97c7d883ceff4

- untagging
  koji list-tagged eln rubygems
  koji untag-build eln rubygems-3.1.4-200.eln102 rubygems-3.1.4-201.eln103

All the rest is automatic.

**Comment by voxik** [comment]
Thx.

I don't need to have privileges, but I want to understand how things are done. E.g. the https://github.com/fedora-ci/eln-periodic repo seems to provide useful insights into how ELN is produced and I was not aware about it. Nobody ever shared this repo AFAIK.

**Comment by voxik** [comment]
How can I check this has propagated everywhere it should (seeing rubygems still listed [here](https://tiny.distro.builders/view--view-eln--x86_64.html))?

**Comment by tdawson** [comment]
Since this is a source rpm that we want removed, you need to look at the source rpm output.
So look here
https://tiny.distro.builders/view-source-package-name-list--view-eln--x86_64.txt
and here
https://tiny.distro.builders/view-buildroot-source-package-name-list--view-eln--x86_64.txt

It looks like it is still in the buildroot list.
I was hoping that removing it from the ELN repo would stop that, but looks like I need to put an excludes in my buildroot repo files as well.

**Comment by voxik** [comment]
And would this help?

https://github.com/minimization/content-resolver/blob/master/config_specs/exclusion.yaml#L60

Or is it not working or for different purpose.

> Side note. Why is there a rubygems source rpm in Fedora, if ruby is also creating rubygems? Shouldn't that be retired?

Since RubyGems version is independent project with independent schedule, it allows easy updates. And as you can see, the independent rubygems package is in newer version then the RubyGems bundled in Ruby ATM, so it is useful.

There is more packages like that.

**Comment by voxik** [comment]
I think we have the similar issue with rubygem-json and rubygem-rake

**Comment by tdawson** [comment]
Yep, looks like those two are also built by ruby.  I'll fix them like I did rubygems.

**Comment by tdawson** [comment]
What about rubygem-power_assert rubygem-rdoc rubygem-test-unit rubygem-minitest  ?
All of them are built by ruby, but also have seperate source rpms.  Should I also exclude them?  Or are they ok to have the other source package in ELN / RHEL9 ?

**Comment by voxik** [comment]
> What about rubygem-power_assert rubygem-rdoc rubygem-test-unit rubygem-minitest ?

Yes, right. They are all the same case.

**Comment by tdawson** [comment]
> > What about rubygem-power_assert rubygem-rdoc rubygem-test-unit rubygem-minitest ?
> 
> Yes, right. They are all the same case.

Adding them into the excludes.

**Comment by tdawson** [comment]
OK, everything is excluded, and I've removed them from ELN repos, as well as RHEL9 alpha repo's.  (Give it an hour or three for the compose repos to be regenerated)
The builds for the packages are still there, but they shouldn't be tagged into anything.

**Comment by voxik** [comment]
BTW should I remove "rubygems" from the "unwanted" packages? Probably yes ....

**Comment by tdawson** [comment]
Yes
unwanted packages is currently binaries only
I know there is plans for srpms, but I don't think that's implemented yet.

**Comment by voxik** [comment]
I think I should alsro remove rubygem-bundler from [here](https://github.com/minimization/content-resolver-input/blob/cef400657c349428e32577d47417b6a4ba336c10/configs/sst_cs_apps-unwanted-ruby.yaml#L9-L10), since this is similar case, although it was never imported into ELN, because the independent package is of older version.
